backend/session/models.py

```diff
@@ -137,6 +137,9 @@ def section_from_any_song(self, filter_by={}):
         To ensure appropriate IP restrictions, most rules should use this
         method instead of operating on the playlist directly.
         """
+        if self.playlist is None:
```

Comment:
There are a couple of experiments which "cheat" a bit in that they make a query over the whole database to pull in specific sections. While I agree with enforcing a playlist on the session, we should check that acceptance / production databases don't contain experiments without associated playlists.

Reply:
Maybe we can think about some properties on the base ruleset like 'playlist_required = False' False by default but can be set to True in this specific ruleset. And we could do a check with that property in mind. What do you think? I think we already had an issue about validation of experiments which is kind of related.

Reply:
I think we can document & enforce that experiments should have playlists associated, no need to create exceptions on the rule level as far as I can see. At some point, `experiment.playlist` was set to `blank=True`, which was mainly a result of mild to severe annoyance when filling in an experiment admin form, only to be bounced back because the playlist wasn't filled. Perhaps we could actually improve this bit: instead, auto-fill the first playlist in the database, and warn the user that they should revisit the playlist(s) associated with the experiment.

Reply:
Perhaps it would be nice to discuss this in a separate issue. I do think it would be good to warn the (admin) user that a Playlist doesn't exist or is empty before launching the experiment. I don't know what's the best place and way to do that. I do know that it took me some debugging before I realised that this was the reason my experiment wasn't working correctly as the frontend just shows a generic error message (could not load first round or something). I believe in throwing exceptions early and often. Some experiments need playlists with sections (sometimes with sections with certain tags or groups, or a minimum amount), so I would assume the admin user would be notified of this in the process of creating or updating an exeriment. At the same time, a playlist might get deleted, and in that scenario I would expect the API to throw an error that gets logged somewhere (frontend, backend, Sentry).